mkosi/distributions/gentoo.py

```diff
@@ -211,43 +109,39 @@ def fetch_fix_stage3(self) -> None:
             else:
                 die("profile names changed upstream?")
 
+        stage3_tar_path = state.cache_dir / stage3_tar
+        cls.stage3_cache = stage3_tar_path.with_name(stage3_tar.name).with_suffix(".tmp")
```

Comment:
Instead of adding the ".tmp" suffix, we should use the tar name without any extensions. Also, we should extract to a fixed location regardless of the tar name and get rid of `cls.stage3_cache` and hardcode the location instead.

Reply:
dropped the `.tmp` suffix.

for `cls.stage3_cache`:
`invoke_emerge()`, called from `install_package()` needs to pass `sysroot`, which points to `cls.stage3_cache`. say we hardcode `cls.stage_cache`, a `Path`, to some directory name so that said `sysroot` ends something like `state.cache_dir.joinpath(stage3_dir)`, where `stage3_dir` is set to `"stage3"`, a `str`. doesn't this buy us anything?